addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/service/SmappeeService.java

```diff
@@ -0,0 +1,294 @@
+                String nowUtcMinus20MinMillis = String.valueOf(nowUtcMinus20Min.getMillis());
+
+                // https://app1pub.smappee.net/dev/v1/servicelocation/123/consumption?aggregation=1&from=1388534400000&to=1391212800000
+
+                String responseReadings = GetData("/dev/v1/servicelocation/" + this.serviceLocationId
+                        + "/consumption?aggregation=1&from=" + nowUtcMinus20MinMillis + "&to=" + nowUtcMillis);
+
+                SmappeeDeviceReading readings = gson.fromJson(responseReadings, SmappeeDeviceReading.class);
+
+                // Sum of 5 minutes in Wh so average power is 60/5 x
+
+                for (SmappeeDeviceReadingConsumption consumption : readings.consumptions) {
+                    consumption.consumption = consumption.consumption * 12;
+                    consumption.solar = consumption.solar * 12;
+                }
+
+                logger.debug("smappee'{}' read", new Object[] { this.serviceLocationId });
+
+                return readings;
+
+            } catch (Exception se) {
+                logger.debug("failed to read smappee '{}'", new Object[] { se.getMessage() });
+            }
+        } while (currentTry++ < this.retry);
+
+        return null;
+
+    }
+
+    public boolean initialize() {
+        // get service locations
+        String response = GetData("/dev/v1/servicelocation");
+
+        if (response.isEmpty()) {
+            return false;
+        }
+
+        SmappeeServiceLocationResponse smappeeServiceLocationResponse = gson.fromJson(response,
+                SmappeeServiceLocationResponse.class);
+
+        if (smappeeServiceLocationResponse == null) {
+            return false;
+        }
+
+        for (SmappeeServiceLocation smappeeServiceLocation : smappeeServiceLocationResponse.serviceLocations) {
+            if (smappeeServiceLocation.name.equals(this.serviceLocationName)) {
+                this.serviceLocationId = Integer.toString(smappeeServiceLocation.serviceLocationId);
+                isInitialized = true;
+
+                return true;
+
+            }
+        }
+        return false;
+    }
+
+    public boolean isInitialized() {
+        return isInitialized;
+    }
+
+    private String GetData(String request) {
+        String url = "https://app1pub.smappee.net" + request;
+
+        HttpClient getClient = new HttpClient();
+
+        GetMethod getMethod = new GetMethod(url);
+        getMethod.getParams().setParameter(HttpMethodParams.RETRY_HANDLER, new DefaultHttpMethodRetryHandler(3, false));
+        getMethod.getParams().setParameter(HttpMethodParams.USER_AGENT,
+                "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2; .NET CLR 1.0.3705;)");
+        getMethod.addRequestHeader("Accept", "application/json");
+
+        String accessTokenToInclude = GetAccessToken();
+        if (accessTokenToInclude.isEmpty()) {
+            logger.error("Could not get access token");
```

Comment:
If this error log a bug or a problem with the external service? If it's not a bug it should not log to error. Same question for the other log to errors in this file.

Reply:
refactored code and logged errors like this :

warn -> external service fails (there is something wrong getting an access/refresh token)
error -> bug while parsing the response from smappee api

Reply:
thank you for the review :-)